app/javascript/components/map/MapPopup.vue

```diff
@@ -110,6 +100,16 @@ export default {
       this.randomKey += 1;
       this.data.datasets = [];
 
+      const ids =[]
+      
+      this.chartData.forEach(item => ids.push(item.id))
+
+      this.chartData.forEach(item => {
+        const color = CHART_COLORS[item.id] || "#e3e3e3"
+
+        this.colors.push(color)
+      })
```

Comment:
Just spotted that below this there's another `forEach` on the same array (`this.chartData`). This could be simplified further to just get the colour in that loop, rather than looping once to create an array in state and then looping again and looking up the array in state. I.e. below:

```
      this.chartData.forEach(item => {
        const backgroundColor = CHART_COLORS[item.id] || "#e3e3e3"

        this.data.datasets.push({
          data: [item.count],
          label: item.name,
          backgroundColor
        })
      });
```

Then all the code above becomes redundant and you don't need `colors` in state either

Reply:
yep good point